cs-alibaba/src/main/java/io/cloudslang/content/alibaba/actions/instances/AllocatePublicIPAddress.java

```diff
@@ -0,0 +1,100 @@
+                    @Output(value = Outputs.PUBLIC_IP_ADDRESS, description = PUBLIC_IP_ADDRESS_DESC),
+                    @Output(value = Outputs.EXCEPTION, description = EXCEPTION_DESC)
+            },
+            responses = {
+                    @Response(text = Outputs.SUCCESS, field = Outputs.RETURN_CODE, value = Outputs.SUCCESS_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.RESOLVED, description = SUCCESS_DESC),
+                    @Response(text = Outputs.FAILURE, field = Outputs.RETURN_CODE, value = Outputs.FAILURE_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.ERROR, description = FAILURE_DESC)
+            }
+    )
+    public Map<String, String> execute(@Param(value = ACCESS_KEY_ID, required = true, description = ACCESS_KEY_ID_DESC) final String accessKeyId,
+                                       @Param(value = ACCESS_KEY_SECRET, required = true, encrypted = true, description = ACCESS_KEY_SECRET_ID_DESC) final String accessKeySecret,
+                                       @Param(value = PROXY_HOST, description = PROXY_HOST_DESC) final String proxyHost,
+                                       @Param(value = PROXY_PORT, description = PROXY_PORT_DESC) final String proxyPort,
+                                       @Param(value = PROXY_USERNAME, description = PROXY_USERNAME_DESC) final String proxyUsername,
+                                       @Param(value = PROXY_PASSWORD, encrypted = true, description = PROXY_PASSWORD_DESC) final String proxyPassword,
+                                       @Param(value = REGION_ID, required = true, description = REGION_ID_DESC) final String regionId,
+                                       @Param(value = INSTANCE_ID, required = true, description = INSTANCE_ID_DESC) final String instanceId) {
+
+        //Get default values and String conversions
+        final String proxyHostImp = defaultIfEmpty(proxyHost, EMPTY);
+        final String proxyPortImp = defaultIfEmpty(proxyPort, DEFAULT_PROXY_PORT);
+        final String proxyUsernameImp = defaultIfEmpty(proxyUsername, EMPTY);
+        final String proxyPasswordImp = defaultIfEmpty(proxyPassword, EMPTY);
+
+        //Validate Inputs
+        Validator validator = new Validator()
+                .validatePort(proxyPortImp, PROXY_PORT);
+
+        if (validator.hasErrors()) {
+            return getFailureResultsMap(validator.getErrors());
+        }
+
+
+        try {
+            final IAcsClient client = ClientUtil.getClient(regionId, accessKeyId, accessKeySecret);
+            final String publicIPAddress = allocatePublicIpAddress(proxyHostImp, proxyPortImp, proxyUsernameImp, proxyPasswordImp, instanceId, client);
+
+
+            final Map<String, String> resultMap = getSuccessResultsMap(ALLOCATE_PUBLIC_IP_ADDRESS_SUCCESS);
+            resultMap.put(PUBLIC_IP_ADDRESS, publicIPAddress);
+
+            return resultMap;
```

Comment:
I think you need to populate return_result and exceptions results as well

Reply:
getSuccessResultsMap & getFailureResultsMap populate returnResult,resultCode and exception if required